File: online-bookstore-server/microservice-bookstore-user/src/main/java/com/onlinebookstore/entity/otherserver/BookResource.java
package com.onlinebookstore.entity.otherserver;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.onlinebookstore.util.BookConstantPool;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.io.Serializable;
import java.util.Date;

/**
 * @author rkc
 * @date 2020/9/17 15:57
 * @version 1.0
 */
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class BookResource implements Serializable {

    /*
    自增长id
     */
    private Integer id;

    /*
    图书外键字段
     */
    @JsonProperty("book_id")
    private Integer bookId;

    /*
    资源路径
     */
    @JsonProperty("resource_url")
    private String resourceUrl;

    /*
    标识，用于区别该资源的作用
     */
    private String symbol;

    /*
    补充说明
     */
    private String supplement;

    @JsonProperty("create_time")
    @JsonFormat(pattern = BookConstantPool.TIME_FORMAT, timezone = BookConstantPool.TIMEZONE)
    private Date createTime;
}

File: online-bookstore-android/app/src/main/java/com/rkc/onlinebookstore/model/book/BookResource.kt
package com.rkc.onlinebookstore.model.book

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

/**
 * @author rkc
 * @date 2020/11/11 18:33
 * @version 1.0
 */
@Parcelize
data class BookResource(
    @SerializedName("id") val id: Int,
    @SerializedName("book_id") val bookId: Int,
    @SerializedName("resource_url") val resourceUrl: String,
    @SerializedName("symbol") val symbol: String,
    @SerializedName("supplement") val supplement: String,
    @SerializedName("create_time") val createTime: Date
) : Parcelable